assets/js/googlesitekit/notifications/datastore/notifications.js

```diff
@@ -402,8 +517,58 @@ export const controls = {
 	),
 };
 
+// eslint-disable-next-line complexity
 export const reducer = createReducer( ( state, { type, payload } ) => {
 	switch ( type ) {
+		case INSERT_NOTIFICATION_INTO_RESOLVED_QUEUE: {
+			const { id } = payload;
+
+			/**
+			 * The notification we want to add to the already-resolved queue.
+			 */
+			const notification = state.notifications?.[ id ];
+
+			if ( notification === undefined ) {
+				global.console.warn(
+					`Could not add notification with ID "${ id }" to queue. Notification "${ id }" is not registered.`
+				);
+
+				break;
+			}
+
+			state.queuedNotifications[ notification.groupID ] =
+				state.queuedNotifications[ notification.groupID ] || [];
+
+			// If the notification is already in the queue, we don't need to add it
+			// again.
+			if (
+				state.queuedNotifications[ notification.groupID ].some(
+					( notificationInQueue ) => notificationInQueue.id === id
+				)
+			) {
+				break;
+			}
+
+			// Find the next notification in the queue that has a lower priority than
+			// the one we're adding, and add this notification ahead of it.
+			const positionForNewNotification = state.queuedNotifications[
+				notification.groupID
+			].findIndex( ( notificationInQueue ) => {
+				return notificationInQueue.priority >= notification.priority;
+			} );
+
+			state.queuedNotifications[ notification.groupID ].splice(
+				positionForNewNotification !== -1
+					? positionForNewNotification
+					: state.queuedNotifications[ notification.groupID ].length +
+							1,
+				0,
+				notification
+			);
```

Comment:
Can we add test coverage to show that this works as intended?

While it may be unlikely, there may be a risk that the notification does not land in the right place if registered while the queue is being populated.